src/System.Resources.Extensions/src/System/Resources/Extensions/TypeNameComparer.cs

```diff
@@ -0,0 +1,123 @@
+
+            ReadOnlySpan<char> typeSpan1 = assemblyQualifiedTypeName1.AsSpan().TrimStart(s_whiteSpaceChars);
+            ReadOnlySpan<char> typeSpan2 = assemblyQualifiedTypeName2.AsSpan().TrimStart(s_whiteSpaceChars);
+
+            // First, compare type names
+            ReadOnlySpan<char> type1 = ReadTypeName(typeSpan1);
+            ReadOnlySpan<char> type2 = ReadTypeName(typeSpan2);
+            if (!type1.Equals(type2, StringComparison.Ordinal))
+                return false;
+
+            // skip separator and whitespace
+            typeSpan1 = typeSpan1.Length > type1.Length ? typeSpan1.Slice(type1.Length + 1).TrimStart(s_whiteSpaceChars) : ReadOnlySpan<char>.Empty;
+            typeSpan2 = typeSpan2.Length > type2.Length ? typeSpan2.Slice(type2.Length + 1).TrimStart(s_whiteSpaceChars) : ReadOnlySpan<char>.Empty;
+
+            // Now, compare assembly simple names ignoring case
+            ReadOnlySpan<char> simpleName1 = ReadAssemblySimpleName(typeSpan1);
+            ReadOnlySpan<char> simpleName2 = ReadAssemblySimpleName(typeSpan2);
+
+            // Don't allow assembly name without simple name portion
+            if (simpleName1.IsEmpty && !typeSpan1.IsEmpty ||
+                simpleName2.IsEmpty && !typeSpan2.IsEmpty)
+                return false;
+
+            // if both are missing assembly name, or either is missing 
+            // assembly name and the other is mscorlib
+            if (simpleName1.IsEmpty)
+                return (simpleName2.IsEmpty || IsMscorlib(simpleName2));
+            if (simpleName2.IsEmpty)
+                return IsMscorlib(simpleName1);
+
+            if (!simpleName1.Equals(simpleName2, StringComparison.OrdinalIgnoreCase))
+                return false;
+
+            // both are mscorlib, ignore culture and key
+            if (IsMscorlib(simpleName1))
+                return true;
+
+            // both have a matching assembly name parse it to get remaining portions
+            // to compare culture and public key token
+            // the following results in allocations.
+            AssemblyName an1 = new AssemblyName(typeSpan1.ToString());
+            AssemblyName an2 = new AssemblyName(typeSpan2.ToString());
+
+            if (an1.CultureInfo?.LCID != an2.CultureInfo?.LCID)
+                return false;
+
+            byte[] pkt1 = an1.GetPublicKeyToken();
+            byte[] pkt2 = an2.GetPublicKeyToken();
+            return pkt1.AsSpan().SequenceEqual(pkt2);
+        }
+
+        public int GetHashCode(string assemblyQualifiedTypeName)
+        {
+            // non-allocating GetHashCode that hashes the type name portion of the string
+            ReadOnlySpan<char> typeSpan = assemblyQualifiedTypeName.AsSpan().TrimStart(s_whiteSpaceChars);
+            ReadOnlySpan<char> typeName = ReadTypeName(typeSpan);
+            ReadOnlySpan<byte> typeNameBytes = MemoryMarshal.AsBytes(typeName);
+
+            return Marvin.ComputeHash32(typeNameBytes, Marvin.DefaultSeed);
```

Comment:
>Marvin [](start = 19, length = 6)

As the comparer is internal, do we need to use Marvin in hashing? what would be the security concern if we didn't use Marvin?

Reply:
I chose this since I needed a reusable hashing function that could work on a span.  3.0 has an [API on String](https://github.com/dotnet/coreclr/pull/20422/files) that does this, but I need to run on netstandard2.0.  I could write my own hashing algo, but I thought it'd be better to use one we were already happy with.

I could just substring and call gethashcode on the string instead, but it seemed easy enough to avoid the allocation by pulling in Marvin.

Reply:
Marvin is already slower than normal hashing algorithms. but if you sub-string and call gethashcode will also call Marvin from there. I was just trying to kill dependency on Marvin code from this assembly. Is it expected GetHashCode will get called many times in our scenarios?

Reply:
It will be called once per call to AddResource, so not a ton.  I suppose I can just do a substring instead.  This is probably all premature optimization.

Reply:
Looks like I can use HashHelpers instead.  That's a much smaller dependency and should do the work I need and be faster.

Reply:
That is reasonable. Thanks.